### PR TITLE
Updated For VS 2015, and fixed few bugs

### PR DESCRIPTION
**Pull request by krssrb**

### CODE REVIEW

File: sql/gameserver_dev_Full.sql [hunk 1]
```diff
@@ -769,79 +769,79 @@ CREATE TABLE IF NOT EXISTS `npc` (
...
-  `skill64` int(11) NOT NULL DEFAULT '0',
-  `skill65` int(11) NOT NULL DEFAULT '0',
-  `skill66` int(11) NOT NULL DEFAULT '0',
-  `skill67` int(11) NOT NULL DEFAULT '0',
-  `skill68` int(11) NOT NULL DEFAULT '0',
-  `skill69` int(11) NOT NULL DEFAULT '0',
-  `skill70` int(11) NOT NULL DEFAULT '0',
-  `skill71` int(11) NOT NULL DEFAULT '0',
-  `skill72` int(11) NOT NULL DEFAULT '0',
+  `skill0` tinyint(4) unsigned NOT NULL DEFAULT '0',
```
Comment (damuras) [inline sql/gameserver_dev_Full.sql:772]: why the change to tinyint(4)

File: sql/gameserver_dev_Full.sql [hunk 2]
```diff
@@ -543,7 +543,7 @@ CREATE TABLE `inventory` (
 `slot264item` int(10) unsigned NOT NULL DEFAULT '0',`slot264qty` int(10) unsigned NOT NULL DEFAULT '0',
 `slot265item` int(10) unsigned NOT NULL DEFAULT '0',`slot265qty` int(10) unsigned NOT NULL DEFAULT '0',
 `slot266item` int(10) unsigned NOT NULL DEFAULT '0',`slot266qty` int(10) unsigned NOT NULL DEFAULT '0',
--- Weapon drawler
+-- Weapon drawer
```
Comment (damuras) [inline sql/gameserver_dev_Full.sql:546]: good catch

File: sql/gameserver_dev_Full.sql [hunk 3]
```diff
@@ -48,73 +48,73 @@ CREATE TABLE `abilities` (
...
 DROP TABLE IF EXISTS `characters`;
 CREATE TABLE `characters` (
-  `id` int(11) NOT NULL AUTO_INCREMENT,
+  `id` int(11) unsigned NOT NULL AUTO_INCREMENT,
   `name` varchar(64) NOT NULL,
   `lastname` varchar(64) NOT NULL,
   `slotID` tinyint(4) NOT NULL,
-  `userID` int(11) NOT NULL,
-  `gender` tinyint(4) NOT NULL,
+  `userID` int(11) unsigned NOT NULL,
```
Comment (damuras) [inline sql/gameserver_dev_Full.sql:55]: why unsigned?

File: sql/gameserver_dev_Full.sql (hunk 1 above)
Comment (krssrb) [inline sql/gameserver_dev_Full.sql:772]: no need for 4 bytes, it can be placed all in 1 byte, it's only 0 and 1, (have skill or not)

File: src/inventory.cpp [hunk 4]
```diff
@@ -652,8 +652,8 @@ void _cb_item_recv_RequestWeaponReload_actionUpdate(mapChannel_t* mapChannel, ac
 			if (itemAmmo->itemTemplate->item.classId == ammoClassId)
 			{
 				// consume ammo
-				sint32 ammoToGrab = min(item->itemTemplate->weapon.clipSize - foundAmmoAmount, itemAmmo->stacksize);
-				foundAmmoAmount += ammoToGrab;
+				sint32 ammoToGrab = min(item->itemTemplate->weapon.clipSize - foundAmmoAmount - item->weaponData.ammoCount, itemAmmo->stacksize);
+				foundAmmoAmount = ammoToGrab + item->weaponData.ammoCount;
 				inventory_reduceStackCount(client, itemAmmo, ammoToGrab);
```
Comment (krssrb) [inline src/inventory.cpp:657]: Fixed bug where on reload weapon, every time consumed max clip size of ammo.
Now only use how much iss missing.

**Comment by Dahrkael** [comment]
this is a preference, more than a bug. if you reload a weapon in the middle of a battle, you are throwing away the whole clip, not just filling it with the 3 bullets missing.
This way you force players to choose between having the full magazine ready to rock or be conservative with scarce ammunition

**Comment by krssrb** [comment]
hmm i dont remember how reaload weapon was working in game, but i know that todays game's refill only missing ammo. We can discus with others members about this on forum

**Comment by damuras** [comment]
I say we let them keep their bullets for now, however this is a good way to up the difficulty if needed when balancing.

**Comment by damuras** [comment]
If @Dahrkael is ok then let's merge it in.
